torch/export/_trace.py

```diff
@@ -678,53 +721,98 @@ def make_argument_spec(i, node) -> ArgumentSpec:
                 f"while writing the metadata for exported program"
             )
 
-    is_joint = graph_signature.backward_signature is not None
+    # TODO(pianpwk): maybe rely on AOTAutograd for unifying this? Very ugly right now
+    if is_training:
```

Comment:
@tugsbayasgalan do you know if we can rely on AOTAutograd (e.g. run_functionalized_fw_and_collect_metadata) to get this signature? It seems like we might(?) need to rely on it to handle some errors in mutation + aliasing too. Not sure if having this make_fx() call within this file is the right thing to do.